lib/matplotlib/animation.py

```diff
@@ -337,6 +347,7 @@ def grab_frame(self, **savefig_kwargs):
         verbose.report('MovieWriter.grab_frame: Grabbing frame.',
                        level='debug')
         try:
+            self._adjust_frame_size()
```

Comment:
What if the animation callback function calls `set_size_inches`? This happens indirectly with any animation using a yt plot when the figure gets updated. yt assumes it can control the figure size, which is a reasonable assumption everywhere else in matplotlib, except for here when it's trying to work around this h264 issue.